content/blog/2016/2016-05-19-update-plugin-for-pipeline-tutorial.md

```diff
@@ -0,0 +1,189 @@
+### How'd I end up here?
+
+Recently, I started working on a project to automate some performance tests for my company's products.  We use the awesome [Gatling](http://gatling.io/#/) load testing tool for these tests, but we've largely been handling the testing very manually to date, due to a lack of bandwidth to get them automated in a clean, maintainable, extensible way.  We have a years-old Jenkins server where we use the [gatling jenkins plugin](https://wiki.jenkins-ci.org/display/JENKINS/Gatling+Plugin) to track the history of certain tests over time, but the setup of the Jenkins instance was very delicate and not easy to reproduce, so it had fallen into a state of disrepair.
+
+Over the last few days I've been putting some effort into getting things more automated and repeatable so that we can really maximize the value that we're getting out of the performance tests.  With some encouragement from the fine folks in the #jenkins IRC channel, I ended up exploring the [JobDSL plugin](https://wiki.jenkins-ci.org/display/JENKINS/Job+DSL+Plugin) plugin and the new [Pipeline jobs](https://jenkins.io/doc/pipeline/).  Combining those two things with some Puppet code to provision a Jenkins server via the [jenkins puppet module](https://github.com/jenkinsci/puppet-jenkins) gave me a really nice way to completely automate my Jenkins setup and get a seed job in place that would create my perf testing jobs.  And the Pipeline job format is just an *awesome* fit for what I wanted to do in terms of being able to easily monitor the stages of my performance tests, and to make the job definitions modular so that it would be really easy to create new performance testing jobs with slight variations.
+
+So everything's going GREAT up to this point.  I'm really happy with how it's all shaping up.  But then... (you knew there was a "but" coming, right?) I started trying to figure out how to add the Gatling Jenkins plugin to the Pipeline jobs, and kind of ran into a wall.  As best as I could tell from my googling, the plugin was probably going to require some modifications in order to be able to be used with Pipeline jobs, but I wasn't able to find any really cohesive documentation that definitively confirmed that or explained how everything fits together.
+
+After a few days of rando n00b mails to the mailing lists, asking dumb questions in IRC, absorbing bits of knowledge from READMEs in various github repos, and just poking at source code, I eventually got it all sorted out.  So, in hopes of saving the next person a little time, and encouraging plugin authors to invest the time to get their plugins working with Pipeline, here are some notes about what I learned.
+
+Spoiler: if you're just interested in looking at the individual git commits that I made on may way to getting the plugin working with Pipeline, have a look at [this github branch](https://github.com/cprice404/gatling-plugin/commits/feature/master/compatibility-with-jenkins-pipeline.individual-commits).
```

Comment:
If possible, it might be best to link to the change in the jenkinsci namespace, rather than your personal GitHub space. Best to minimize the risk that the link will break over time.